### DISCUSSION TITLE
懒加载树，按方向键（left-arrow/right-arrow）为什么会触发 node-click 事件 ？

### DISCUSSION THREAD

**Discussion by WestFarmer** [element-plus/element-plus#16321] [Q&A]
在懒加载模式下，左右方向键是关闭/展开一个节点，我想在节点选中时执行我的代码（鼠标点击节点，节点获得焦点时按回车键）

```
<el-tree
        lazy
        :load="load"
        :node-key="props.treeProps.value"
        :current-node-key="value"
        :props="props.treeProps"
        :expand-on-click-node="false"
        @node-click="nodeClick"
        @current-change="currentChange"
      />

const nodeClick = (a: any, b: any, c: any, e: any) => {
  // TODO why left/right arrow key navigation also trigger this event ?
  value.value = a[props.treeProps.value]
  label.value = a[props.treeProps.label]
  dialog.value.visible = false
}
```
1. 怎么实现按回车键改变节点的选中状态？
2. 如果避免左右方向键触发节点选中事件？

求好心人帮帮忙

**Comment by Aseisman** [comment]
https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGRpdiBAa2V5ZG93bi5sZWZ0LmNhcHR1cmU9XCJrZXlDbGlja1wiIEBrZXlkb3duLnJpZ2h0LmNhcHR1cmU9XCJrZXlDbGlja1wiPlxuICAgIDxlbC10cmVlIHJlZj1cImVsVHJlZVJlZlwiIHN0eWxlPVwibWF4LXdpZHRoOiA2MDBweFwiIDpkYXRhPVwiZGF0YVwiIDpwcm9wcz1cImRlZmF1bHRQcm9wc1wiIEBub2RlLWNsaWNrPVwiaGFuZGxlTm9kZUNsaWNrXCIgLz5cbiAgPC9kaXY+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSBcInZ1ZVwiO1xuXG5pbnRlcmZhY2UgVHJlZSB7XG4gIGxhYmVsOiBzdHJpbmdcbiAgY2hpbGRyZW4/OiBUcmVlW11cbn1cbmNvbnN0IGVsVHJlZVJlZiA9IHJlZigpO1xuY29uc3Qga2V5Q2xpY2sgPSAoZSkgPT4ge1xuICAvLyDlnKjmraTlpITnm5HlkKzlm57ovabplK7lkozlt6blj7PplK5cbiAgZS50YXJnZXQuY2xpY2soKTtcbiAgZS5zdG9wUHJvcGFnYXRpb24oKTtcbn1cbmNvbnN0IGhhbmRsZU5vZGVDbGljayA9IChhOiBhbnksIGI6IGFueSwgYzogYW55LCBlOiBhbnkpID0+IHtcbiAgY29uc29sZS5sb2coYSk7XG59XG5jb25zdCBkYXRhOiBUcmVlW10gPSBbXG4gIHtcbiAgICBsYWJlbDogJ0xldmVsIG9uZSAxJyxcbiAgICBjaGlsZHJlbjogW1xuICAgICAge1xuICAgICAgICBsYWJlbDogJ0xldmVsIHR3byAxLTEnLFxuICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgIHtcbiAgICAgICAgICAgIGxhYmVsOiAnTGV2ZWwgdGhyZWUgMS0xLTEnLFxuICAgICAgICAgIH0sXG4gICAgICAgIF0sXG4gICAgICB9LFxuICAgIF0sXG4gIH0sXG4gIHtcbiAgICBsYWJlbDogJ0xldmVsIG9uZSAyJyxcbiAgICBjaGlsZHJlbjogW1xuICAgICAge1xuICAgICAgICBsYWJlbDogJ0xldmVsIHR3byAyLTEnLFxuICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgIHtcbiAgICAgICAgICAgIGxhYmVsOiAnTGV2ZWwgdGhyZWUgMi0xLTEnLFxuICAgICAgICAgIH0sXG4gICAgICAgIF0sXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICBsYWJlbDogJ0xldmVsIHR3byAyLTInLFxuICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgIHtcbiAgICAgICAgICAgIGxhYmVsOiAnTGV2ZWwgdGhyZWUgMi0yLTEnLFxuICAgICAgICAgIH0sXG4gICAgICAgIF0sXG4gICAgICB9LFxuICAgIF0sXG4gIH0sXG4gIHtcbiAgICBsYWJlbDogJ0xldmVsIG9uZSAzJyxcbiAgICBjaGlsZHJlbjogW1xuICAgICAge1xuICAgICAgICBsYWJlbDogJ0xldmVsIHR3byAzLTEnLFxuICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgIHtcbiAgICAgICAgICAgIGxhYmVsOiAnTGV2ZWwgdGhyZWUgMy0xLTEnLFxuICAgICAgICAgIH0sXG4gICAgICAgIF0sXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICBsYWJlbDogJ0xldmVsIHR3byAzLTInLFxuICAgICAgICBjaGlsZHJlbjogW1xuICAgICAgICAgIHtcbiAgICAgICAgICAgIGxhYmVsOiAnTGV2ZWwgdGhyZWUgMy0yLTEnLFxuICAgICAgICAgIH0sXG4gICAgICAgIF0sXG4gICAgICB9LFxuICAgIF0sXG4gIH0sXG5dXG5cbmNvbnN0IGRlZmF1bHRQcm9wcyA9IHtcbiAgY2hpbGRyZW46ICdjaGlsZHJlbicsXG4gIGxhYmVsOiAnbGFiZWwnLFxufVxuPC9zY3JpcHQ+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXSxcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJfbyI6e319

**Reply by Aseisman** [reply]
参照的源码来进行捕获键盘事件的，可以看看
